Assets/Scripts/PlayerIcon.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UI;
+
+/// <summary>
+/// Manages Player Icon Location
+/// Appears when camera switch to map mode and when in isometric view
+/// 
+/// </summary>
+public class PlayerIcon : MonoBehaviour
```

Comment:
This entire class seems over-engineered when it could simply have an enable icon and a disable icon method that subscribes to the events that turns it on and off.